connector_search_engine/models/se_binding.py

```diff
@@ -60,12 +60,28 @@ def recompute_json(self, force_export=False):
             mapper = work.component(usage='se.export.mapper')
             lang = work.index.lang_id.code
             for record in work.records.with_context(lang=lang):
-                data = mapper.map_record(record).values()
-                if record.data != data or force_export:
-                    vals = {'data': data}
-                    if record.sync_state in ('done', 'new'):
-                        vals['sync_state'] = 'to_update'
-                    record.write(vals)
+                values = record._get_values_recompute_json(
+                    mapper, force_export=force_export)
+                if values:
+                    record.write(values)
+
+    @api.multi
+    def _get_values_recompute_json(self, mapper, force_export=False):
+        """
+        Get values for current recordset during recompute_json
+        :param mapper: export mapper
+        :param force_export: bool
+        :return: dict
+        """
+        self.ensure_one()
+        data = mapper.map_record(self).values()
+        if self.data != data or force_export:
```

Comment:
hum I think we will have an issue here.
Indeed the comparison between the new data and the existing one is done in the method.
In the case that you inherit this method to update the data (ex : the default value for the SEO title)
As the data have being tested here you change in the data will not impact the status.

Moreover if we want to update the data of the mapping we can already do it by adding a custom mapping so I do not think that we need to add a hook what do you think ? @lmignon @Cedric-Pigeon @acsonefho

Reply:
Hi @sebastienbeau 
I update the function to keep your logic with the sync_state and the data.
For the case of seo_title, we want to update it in the data but also into Odoo (so the field value).
So here the 2 `if` will detect changes on `data` and the `values` will also contains the value of `seo_title` (to update it if necessary).